app/src/main/java/de/tum/in/tumcampusapp/fragments/BarrierfreeFacilityListFragment.java

```diff
@@ -0,0 +1,97 @@
+import de.tum.in.tumcampusapp.tumonline.TUMBarrierFreeRequest;
+import de.tum.in.tumcampusapp.tumonline.TUMRoomFinderRequestFetchListener;
+import se.emilsjolander.stickylistheaders.StickyListHeadersListView;
+
+public class BarrierfreeFacilityListFragment extends Fragment
+        implements TUMRoomFinderRequestFetchListener, AdapterView.OnItemClickListener {
+    private TUMBarrierFreeRequest request;
+    private StickyListHeadersListView stickyList;
+    private RoomFinderListAdapter adapter;
+    private int pageID;
+    List<Map<String, String>> facilities;
+
+    private RecentsManager recentsManager;
+
+    @Nullable
+    @Override
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_barrierfree_facility_list, container, false);
+        stickyList = (StickyListHeadersListView) view.findViewById(R.id.activity_barrier_free_facilities_list_view);
+        request = new TUMBarrierFreeRequest(getActivity());
+
+        recentsManager = new RecentsManager(getActivity(), RecentsManager.ROOMS);
+
+        if (getArguments().containsKey(Const.BARRIER_FREE_FACILITY_PAGE_ID)) {
+            pageID = getArguments().getInt(Const.BARRIER_FREE_FACILITY_PAGE_ID);
+        }
+
+        switch (pageID){
+            case 0:
+                request.fetchListOfNerbyFacilities(getActivity(), this);
+                break;
+            case 1:
+                request.fetchListOfToilets(getActivity(), this);
+                break;
+            case 2:
+                request.fetchListOfElevators(getActivity(),this);
+                break;
+            default:
+                break;
+        }
+
+        return view;
+    }
+
+    @Override
+    public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+        Map<String, String> facility = facilities.get(position);
+
+        StringBuilder val = new StringBuilder();
+        Bundle b = new Bundle();
+        for (Map.Entry<String, String> entry : facility.entrySet()) {
+            val.append(entry.getKey()).append('=').append(entry.getValue()).append(';');
+            b.putString(entry.getKey(), entry.getValue());
+        }
+        recentsManager.replaceIntoDb(val.toString());
+
+        Intent intent = new Intent(getActivity(), RoomFinderDetailsActivity.class);
+        intent.putExtra(RoomFinderDetailsActivity.EXTRA_ROOM_INFO, b);
+        startActivity(intent);
+    }
+
+    @Override
+    public void onFetch(List<Map<String, String>> result) {
+        facilities = result;
+        adapter = new RoomFinderListAdapter(getActivity(), facilities);
+        stickyList.setAdapter(adapter);
+        stickyList.setOnItemClickListener(this);
+    }
+
+    @Override
+    public void onFetchError(String errorReason) {
+        request.cancelRequest(true);
+    }
+
+    @Override
+    public void onNoInternetError() {
```

Comment:
They are from the interface TUMRoomFinderRequestFetchListener and I have to implement them